File: functions/libsodium/variants/go/3_b64_urlsafe_char_to_byte_4.go
package main

func b64_urlsafe_char_to_byte(c int32) uint {
    if unicode.IsUpper(c) {
        return uint(c - 'A')
    }
    if unicode.IsLower(c) {
        return uint(c - 'a' + 26)
    }
    if unicode.IsDigit(c) {
        return uint(c - '0' + 52)
    }
    if c == '-' {
        return 62
    }
    if c == '_' {
        return 63
    }
    return 0
}


func main(){
    b64_urlsafe_char_to_byte(0)
}

File: functions/libsodium/variants/c/3_b64_urlsafe_char_to_byte_4.c
#include <stdint.h>

static unsigned int b64_urlsafe_char_to_byte(int c)
{
    if (('A' <= c) && (c <= 'Z'))
        return c - 'A';
    if (('a' <= c) && (c <= 'z'))
        return c - 'a' + 26;
    if (('0' <= c) && (c <= '9'))
        return c - '0' + 52;
    if (c == '-')
        return 62;
    return (c == '_') ? 63 : 0;
}


int main(){
    b64_urlsafe_char_to_byte(0);
}
